sharding/xorValidatorShuffler.go

```diff
@@ -249,30 +240,27 @@ func (rxs *randXORShuffler) computeNewShards(
 // shuffleOutNodes shuffles the list of eligible validators in each shard and returns the array of shuffled out
 // validators
 func shuffleOutNodes(
-	eligible map[uint32][]Validator,
-	waiting map[uint32][]Validator,
+	eligible [][]Validator,
+	waiting [][]Validator,
 	leaving []Validator,
 	randomness []byte,
-) ([]Validator, map[uint32][]Validator, []Validator) {
+	numShardsPlusMeta uint32,
+) ([]Validator, [][]Validator, []Validator) {
 	shuffledOut := make([]Validator, 0)
-	newEligible := make(map[uint32][]Validator)
+	newEligible := make([][]Validator, numShardsPlusMeta)
 	var shardShuffledOut []Validator
 
-	for shard := uint32(0); shard < uint32(len(eligible)-1); shard++ {
+	for shard := uint32(0); shard < numShardsPlusMeta; shard++ {
 		validators := eligible[shard]
```

Comment:
same problem as stated above if the numShardsPlusMeta is larger than len(eligible) this will break.

Reply:
undone. not the case any more